OneCommunityGlobal/HighestGoodNetworkApp
# XiaoW_Refactor of time entry controller and fix edit history issue

# Description
This PR along with the backend changes aims to refactor the time entry controller for post, edit, and delete time entries. They also fix the edit time entry history not added to history issue. 

## Related PRS (if any):
This frontend PR is related to the [#881](https://github.com/OneCommunityGlobal/HGNRest/pull/881#issue-2234712272) backend PR.
…

## Main changes explained:
- integrated all the changes to `userprofile` caused by time entry operation into the backend (such as `updateHoursByCategory` and `editHoursByCategory` in TimeEntryForm component)
- refactor the logic
- remove unused files/logic


## How to test:
1. check into current branch and backend PR #[881](https://github.com/OneCommunityGlobal/HGNRest/pull/881#issue-2234712272)
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin user -> test any operations on time entry (create, edit, delete, change others'...)
5. log as vol user -> test any operation on time entry.
